k2/csrc/fsa_algo_test.cu

```diff
@@ -340,7 +340,19 @@ TEST(FsaAlgo, ShortestPath) {
     FsaVec fsa_vec = CreateFsaVec(3, &fsa_array[0]);
     fsa_vec = fsa_vec.To(context);
 
-    Ragged<int32_t> best_path_arc_indexes = ShortestPath(fsa_vec);
+    Ragged<int32_t> state_batches = GetStateBatches(fsa_vec, true);
+    Array1<int32_t> dest_states = GetDestStates(fsa_vec, true);
+    Ragged<int32_t> incoming_arcs = GetIncomingArcs(fsa_vec, dest_states);
+    Ragged<int32_t> entering_arc_batches =
+        GetEnteringArcIndexBatches(fsa_vec, incoming_arcs, state_batches);
+
+    bool log_semiring = false;
+    Array1<int32_t> entering_arcs;
+    GetForwardScores<float>(fsa_vec, state_batches, entering_arc_batches,
```

Comment:
should we create a wrapper function to get `forward_scores` and `entering_arcs`? I just think maybe we never use above values (e.g. `state_batches`, `dest_states`...) directly in most of calling place, they are just used to call `GetForwardScores`

Reply:
The advantage of calling them separately is that `state_batches` can be reused
across different calls, e..g, `GetForwardScores`,  `GetEnteringArcIndexBatches`, etc.

Reply:
I didn't mean to put them in `GetForwardScores` or `ShortestPath`, I just mean that we can define a helper function to wrap those things, then we can just call this helper function in the case where we just want to get `forward_scores` or `entering_arcs`;